wgpu-core/src/device/ray_tracing.rs

```diff
@@ -0,0 +1,407 @@
+            max_instance_count: desc.max_instances,
+        })
+    }
+}
+
+impl<G: GlobalIdentityHandlerFactory> Global<G> {
+    pub fn device_create_blas<A: HalApi>(
+        &self,
+        device_id: id::DeviceId,
+        desc: &resource::BlasDescriptor,
+        sizes: wgt::BlasGeometrySizeDescriptors,
+        id_in: Input<G, BlasId>,
+    ) -> (BlasId, Option<u64>, Option<CreateBlasError>) {
+        profiling::scope!("Device::create_blas");
+
+        let hub = A::hub(self);
+        let mut token = Token::root();
+        let fid = hub.blas_s.prepare(id_in);
+
+        let (device_guard, mut token) = hub.devices.read(&mut token);
+        let error = loop {
+            let device = match device_guard.get(device_id) {
+                Ok(device) => device,
+                Err(_) => break DeviceError::Invalid.into(),
+            };
+            #[cfg(feature = "trace")]
+            if let Some(ref trace) = device.trace {
+                trace.lock().add(trace::Action::CreateBlas {
+                    id: fid.id(),
+                    desc: desc.clone(),
+                    sizes: sizes.clone(),
+                });
+            }
+
+            let blas = match device.create_blas(device_id, desc, sizes) {
+                Ok(blas) => blas,
+                Err(e) => break e,
+            };
+            let handle = blas.handle;
+
+            let ref_count = blas.life_guard.add_ref();
+
+            let id = fid.assign(blas, &mut token);
+            log::info!("Created blas {:?} with {:?}", id, desc);
+
+            device.trackers.lock().blas_s.insert_single(id, ref_count);
+
+            return (id.0, Some(handle), None);
+        };
+
+        let id = fid.assign_error(desc.label.borrow_or_default(), &mut token);
+        (id, None, Some(error))
+    }
+
+    pub fn device_create_tlas<A: HalApi>(
+        &self,
+        device_id: id::DeviceId,
+        desc: &resource::TlasDescriptor,
+        id_in: Input<G, TlasId>,
+    ) -> (TlasId, Option<CreateTlasError>) {
+        profiling::scope!("Device::create_tlas");
+
+        let hub = A::hub(self);
+        let mut token = Token::root();
+        let fid = hub.tlas_s.prepare(id_in);
+
+        let (device_guard, mut token) = hub.devices.read(&mut token);
+        let error = loop {
+            let device = match device_guard.get(device_id) {
+                Ok(device) => device,
+                Err(_) => break DeviceError::Invalid.into(),
+            };
+            #[cfg(feature = "trace")]
+            if let Some(ref trace) = device.trace {
+                trace.lock().add(trace::Action::CreateTlas {
+                    id: fid.id(),
+                    desc: desc.clone(),
+                });
+            }
+
+            let tlas = match device.create_tlas(device_id, desc) {
+                Ok(tlas) => tlas,
+                Err(e) => break e,
+            };
+            let ref_count = tlas.life_guard.add_ref();
```

Comment:
It appears this is causing the issue, however I think getting this up to work with the arcanization changes are likely to fix this.

Reply:
I am just going to randomley jump into this thread but I had a question. when this does finally release will there be a mimal, *pure* accerated ray tracing option/api? I know it's not the main point but have just ray tracing agaist geometrey without the lighting can be useful for gpu raytraced sound or gpu physics stuff like that. "Ray tracing" as in, tracing rays against triangles, is an increaibley use tool for 3d anything really so being able to take advantage of that hardware acceration for more then just lighting i think should be a priority, but not nesserally the first one. anyway glad to see something is chipping away that this hope it comes to the release version soon :)

Reply:
Our api should end up just as opinionated as the vk/d3d apis.